addOns/pscanrules/src/main/java/org/zaproxy/zap/extension/pscanrules/PiiScanRule.java

```diff
@@ -93,6 +94,11 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         if (ResourceIdentificationUtils.isCss(msg) || ResourceIdentificationUtils.isImage(msg)) {
             return;
         }
+        HttpResponseHeader responseHeader = msg.getResponseHeader();
+        if (!getAlertThreshold().equals(AlertThreshold.LOW)
+                && !isMessageSuitableForNonLowThreshold(responseHeader)) {
+            return;
+        }
         if (!getAlertThreshold().equals(AlertThreshold.LOW) && isPdfMessage(msg)) {
```

Comment:
Better merge these conditions (i.e. `isPdfMessage` should be part of `isMessageSuitableForNonLowThreshold`).